client/autoclient.go

```diff
@@ -0,0 +1,452 @@
+	DumpWIFPrivateKey(ctx context.Context, addr stdaddr.Address) (string, error)
+	VSPFeeHashForTicket(ctx context.Context, ticketHash *chainhash.Hash) (chainhash.Hash, error)
+	UpdateVspTicketFeeToStarted(ctx context.Context, ticketHash, feeHash *chainhash.Hash, host string, pubkey []byte) error
+	GetTransactionsByHashes(ctx context.Context, txHashes []*chainhash.Hash) (txs []*wire.MsgTx, notFound []*wire.InvVect, err error)
+	ReserveOutputsForAmount(ctx context.Context, account uint32, amount dcrutil.Amount, minconf int32) ([]wallet.Input, error)
+	UnlockOutpoint(txHash *chainhash.Hash, index uint32)
+	NewChangeAddress(ctx context.Context, account uint32) (stdaddr.Address, error)
+	RelayFee() dcrutil.Amount
+	SignTransaction(ctx context.Context, tx *wire.MsgTx, hashType txscript.SigHashType, additionalPrevScripts map[wire.OutPoint][]byte,
+		additionalKeysByAddress map[string]*dcrutil.WIF, p2shRedeemScriptsByAddress map[string][]byte) ([]wallet.SignatureError, error)
+	SetPublished(ctx context.Context, hash *chainhash.Hash, published bool) error
+	AddTransaction(ctx context.Context, tx *wire.MsgTx, blockHash *chainhash.Hash) error
+	UpdateVspTicketFeeToPaid(ctx context.Context, ticketHash, feeHash *chainhash.Hash, host string, pubkey []byte) error
+	NetworkBackend() (wallet.NetworkBackend, error)
+	RevokeTickets(ctx context.Context, rpcCaller wallet.Caller) error
+	UpdateVspTicketFeeToErrored(ctx context.Context, ticketHash *chainhash.Hash, host string, pubkey []byte) error
+	AgendaChoices(ctx context.Context, ticketHash *chainhash.Hash) (choices []wallet.AgendaChoice, voteBits uint16, err error)
+	TSpendPolicyForTicket(ticketHash *chainhash.Hash) map[string]string
+	TreasuryKeyPolicyForTicket(ticketHash *chainhash.Hash) map[string]string
+	AbandonTransaction(ctx context.Context, hash *chainhash.Hash) error
+	TxConfirms(ctx context.Context, hash *chainhash.Hash) (int32, error)
+	ForUnspentUnexpiredTickets(ctx context.Context, f func(hash *chainhash.Hash) error) error
+	IsVSPTicketConfirmed(ctx context.Context, ticketHash *chainhash.Hash) (bool, error)
+	UpdateVspTicketFeeToConfirmed(ctx context.Context, ticketHash, feeHash *chainhash.Hash, host string, pubkey []byte) error
+	VSPTicketInfo(ctx context.Context, ticketHash *chainhash.Hash) (*wallet.VSPTicket, error)
+	SignMessage(ctx context.Context, msg string, addr stdaddr.Address) (sig []byte, err error)
+}
+
+type AutoClient struct {
+	Wallet Wallet
+	Policy Policy
+	*Client
+
+	mu   sync.Mutex
+	jobs map[chainhash.Hash]*feePayment
+
+	log slog.Logger
+}
+
+type Config struct {
+	// URL specifies the base URL of the VSP
+	URL string
+
+	// PubKey specifies the VSP's base64 encoded public key
+	PubKey string
+
+	// Dialer specifies an optional dialer when connecting to the VSP.
+	Dialer DialFunc
+
+	// Wallet specifies a loaded wallet.
+	Wallet Wallet
+
+	// Default policy for fee payments unless another is provided by the
+	// caller.
+	Policy Policy
```

Comment:
nit: Can Policy be a pointer type? It doesn't make any functional difference really, just usually we would use a pointer here I think.